ncwriter/template.py

```diff
@@ -247,30 +248,38 @@ def ensure_consistency(self):
                     )
                 )
 
-    def _create_var_opts(self, vdict):
+    def _create_var_opts(self, vname, vdict):
         """Return a dictionary of attributes required for the creation of variable
-        defined by :vdict: This include creation/special options like:
-            `zlib`
-            `least_significant_digit`
-            `fill_value`
-            etc
+        defined by :vdict:. This includes creation/special options like `zlib`,
+        `least_significant_digit`, `fill_value`, etc...
+
+        Also check that only one fill value attribute is provided, and map it to
+        the `fill_value` option for createVariable. If both `fill_value` and `FillValue`
+        are specified:
+        * If they have exactly the same value, print a warning and proceed;
+        * If they have different values, raise ValueError.
+
+        :vname: is only used for reporting the variable name in error/warning message
         """
-        metadata_dict = metadata_attributes(vdict)
         special_dict = special_attributes(vdict)
         struct_keys = self.cattrs.intersection(special_dict.keys())
         fill_aliases = self.fill_aliases.intersection(special_dict.keys())
-        fill_aliases.update(
-            self.fill_aliases.intersection(metadata_dict.keys())  # in case fill value was specified without underscore
-        )
 
         if len(fill_aliases) > 1:
-            raise ValueError('You can only provide one missing value alias!')
+            fill_dict = {f: special_dict[f] for f in fill_aliases}
+            unique_values = set(fill_dict.values())
+            msg = "Multiple fill value options specified for variable '{vname}': {fill_dict}.".format(
+                vname=vname, fill_dict=fill_dict
+            )
+            if len(unique_values) == 1:
+                warn(msg)
+            else:
+                raise ValueError(msg)
```

Comment:
You  might want to make it clear that the problem is that there are multiple distinct values for fill value aliases.